net: openthread: add Time Sync support

edmont opened the pull request:
Implement `otPlatTimeGetXtalAccuracy` which is missing when Time Sync is enabled.

Even if they are not exactly the same for OpenThread I'm leveraging `otPlatRadioGetCslAccuracy` here not to add too much complication in the KConfig. I believe some simplification should be done in OpenThread regarding both functions.

rlubos commented on subsys/net/lib/openthread/platform/alarm.c:
```diff
@@ -126,3 +126,10 @@ uint32_t otPlatAlarmMicroGetNow(void)
 	return (uint32_t)(k_ticks_to_us_floor64(k_uptime_ticks()) - time_offset_us);
 }
+
+#if defined(CONFIG_OPENTHREAD_TIME_SYNC)
```
Can we implement the function unconditionally? If it's not used, the linker will just drop it.

edmont committed:
net: openthread: add Time Sync support

Implement `otPlatTimeGetXtalAccuracy` which is missing when Time
Sync is enabled.

Signed-off-by: Eduardo Montoya <eduardo.montoya@nordicsemi.no>
